app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -294,3 +302,100 @@ def test_hanging_protocol_clientside():
         }
     )
     assert form.is_valid()
+
+
+@pytest.mark.parametrize(
+    "number_of_images,number_of_overlays,expected_help_text",
+    (
+        (
+            0,
+            0,
+            'Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information',
+        ),
+        (
+            1,
+            0,
+            'The following interfaces are used in your {}: test-ci-image-0. Example usage: {{"main": ["test-ci-image-0"]}}. Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information',
+        ),
+        (
+            0,
+            1,
+            'The following interfaces are used in your {}: test-ci-overlay-0. Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information',
+        ),
+        (
+            1,
+            1,
+            'The following interfaces are used in your {}: test-ci-image-0 and test-ci-overlay-0. Example usage: {{"main": ["test-ci-image-0", "test-ci-overlay-0"]}}. Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information',
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "overlay_type",
+    (
+        InterfaceKindChoices.SEGMENTATION,
+        InterfaceKindChoices.HEAT_MAP,
+        InterfaceKindChoices.DISPLACEMENT_FIELD,
+    ),
+)
+@pytest.mark.django_db
+def test_forms_view_content_help_text(
```

Comment:
I put this form test (still not final) under hanging_protocols since it is related to the view_content field of the HangingProtocolMixin, and the logic of assessing the forms (Algorithm, Archive, ReaderStudy, Phase) is similar but not identical. Therefore, the test will be long. Now, I put the test for Algorithm and Archive forms and will discuss first if it is better to put the test all here, or split them over multiple tests or put each test in its relevant app.

Reply:
outdated